controllers/templatesync/template_sync.go

```diff
@@ -165,6 +165,25 @@ func (r *PolicyReconciler) Reconcile(ctx context.Context, request reconcile.Requ
 		return reconcile.Result{}, nil
 	}
 
+	// Check duplicate names in configuration-policies
+	has, err := hasDupName(instance)
+	if err != nil {
+		reqLogger.Error(err, "Failed to check duplicate configurationpolicy name")
+
+		return reconcile.Result{}, nil
+	}
```

Comment:
I think the error flow here could cause some problems. It looks like an error can occur if a template object definition isn't valid and can't be unmarshalled. This return makes it give up forever, without putting a status on the policy...

I think the unmarshal error could be ignored in `hasDupName`, and just skip those templates. So `hasDupName` wouldn't return an error, and the logic would be a bit simpler. The same error should appear later when the templates are being considered individually, right?

Reply:
good great catch!